### ISSUE TITLE
The channel is has been closed When we use more that 1 parallel process

### ISSUE THREAD

**Issue by antonim** [amphp/parallel#9]
Hello. I tried to use this code for getting some data  from url array
```php
Loop::run(function ()  {
    $uris = [
        'http://local.test/sleep.php?sec=1',
        'http://local.test/sleep.php?sec=5',
    ];

    $threads = [];
    foreach ($uris as $key => $uri) {

        $threads[] = Thread::spawn(function ($uri) {
            $result = file_get_contents($uri);
            return $result;
        }, $uri);
    }

    foreach ($threads as $thread) {
        $result[] = yield $thread->join();
    }
});
```

Code from sleep.php : `<?php sleep($_GET['sec']);`

And it is works correctly, but if i change the line order of urls from 
 'http://local.test/sleep.php?sec=1',
 'http://local.test/sleep.php?sec=5',
to 
 'http://local.test/sleep.php?sec=5',
 'http://local.test/sleep.php?sec=1',

Every time I've error:
```txt
Fatal error: Uncaught Amp\Parallel\Sync\ChannelException: The channel is has been closed in /home/anton/www/test/php/parallel/vendor/amphp/parallel/lib/Sync/ChannelledSocket.php:251
```
Could you tell me how I can solve such problem?

**Comment by trowski** [comment]
This was an issue I created in 9cc70c1f86bf8ffa8ca744e40702502410b733f0, but should now have been fixed with a31f3491ca25db50b5a253b30fee2ade2ae6bd9f. Pull the latest master and try again please.

**Comment by antonim** [comment]
Thanks a lot, it works!
